src/Compilers/CSharp/Portable/Syntax/CSharpSyntaxDiffer.cs

```diff
@@ -0,0 +1,23 @@
+﻿namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal sealed class CSharpSyntaxDiffer : SyntaxDiffer
+    {
+        internal CSharpSyntaxDiffer(SyntaxNode oldNode, SyntaxNode newNode, bool computeNewText)
+            : base(oldNode, newNode, computeNewText) { }
+
+        protected override bool AreSimilarCore(SyntaxNode node1, SyntaxNode node2)
```

Comment:
i don't really understand the logic below.  it seems to be saying that Classes/Structs/Interfaces are all similar, and all other types are not similar.  Why si that.  say you have two exactly equivalent other types of nodes... they would not be similar?